python/lsst/ts/externalscripts/base_parameter_march.py

```diff
@@ -0,0 +1,461 @@
+# This file is part of ts_externalscripts
+#
+# Developed for the LSST Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["BaseParameterMarch"]
+
+import abc
+import asyncio
+import types
+
+import numpy as np
+import yaml
+from lsst.ts import salobj
+from lsst.ts.standardscripts.base_block_script import BaseBlockScript
+
+
+class BaseParameterMarch(BaseBlockScript):
+    """Perform a parameter march by incrementally adjusting a
+    specific combination of degrees of freedom. This process
+    involves moving the parameters in specified increments
+    to evaluate the system's response. The sensitivity matrix
+    is an example of this technique, where the system's behavior
+    is assessed by capturing data at various incremental adjustments
+    for each degree of freedom.
+
+
+    Parameters
+    ----------
+    index : `int`
+        Index of Script SAL component.
+
+    Notes
+    -----
+    **Checkpoints**
+
+    * Step 1/n_steps starting position: start_position.
+    * Step n/n_steps.
+    """
+
+    def __init__(self, index, descr="Perform a parameter march.") -> None:
+        super().__init__(index=index, descr=descr)
+
+        self.ocps = None
+
+        self.config = None
+        self.dofs = np.zeros(50)
+
+        self.total_offset = 0.0
+        self.iterations_started = False
+
+    @property
+    @abc.abstractmethod
+    def tcs(self):
```

Comment:
The `tcs` is the same for both cameras, no need to make it an abstract property.